cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js

```diff
@@ -113,45 +109,115 @@ function getPlansForWidget() {
     return plans;
 }
 
+/**
+ * Get feePlans with BO info
+ * @param {Object} feePlans feePlans
+ * @returns {Array} feePlans
+ */
+function getFeePlansBoFormat(feePlans) {
+    var plans = [];
+    feePlans.forEach(function (feePlan) {
+        var feePlanKey = getSitePrefNameFromPlan(feePlan);
+        if (Site.getCurrent()
+            .getCustomPreferenceValue(feePlanKey)) {
+            var plan = feePlan;
+            plan.key = feePlanKey;
+            plan.max_display_amount = Site.getCurrent()
+                .getCustomPreferenceValue(feePlanKey + '_max') * 100;
+            plan.min_display_amount = Site.getCurrent()
+                .getCustomPreferenceValue(feePlanKey + '_min') * 100;
+            plan.payment_method = almaCheckoutHelper.getPlanPaymentMethodID(feePlan);
+            plans.push(plan);
+        }
+    });
+
+    return plans;
+}
+
 /**
  * Get data to initialize widget in cart and product detail
  * @param {string} locale e.g. "fr_FR"
  * @param {dw.order.Basket} currentBasket current basket
  * @returns {array} eligible data
  */
 function getPlansForCheckout(locale, currentBasket) {
-    var plansForEligibility = getAllowedPlans();
+    var feePlans = getFeePlans();
+
```

Comment:
I think this function can be refactored, the code inside can be split in 3 functions at least